src/scene/vcMedia.cpp

```diff
@@ -144,35 +143,106 @@ void vcMedia::AddToScene(vcState *pProgramState, vcRenderData *pRenderData)
+      udFree(m_pImageData);
+      m_imageDataSize = 0;
+
+      if (pData != nullptr)
+      {
+        if (m_image.pTexture != nullptr)
+          vcTexture_GetSize(m_image.pTexture, &imageSize.x, &imageSize.y);
+
+        if ((uint32_t)imageSize.x == width && (uint32_t)imageSize.y == height) // imageSize will not == for null pTexture
+        {
+          vcTexture_UploadPixels(m_image.pTexture, pData, width, height);
+        }
+        else
+        {
+          vcTexture_Destroy(&m_image.pTexture);
+
+          if (vcTexture_Create(&m_image.pTexture, width, height, pData, vcTextureFormat_RGBA8) != udR_Success)
+            m_loadStatus = vcSLS_Failed;
+        }
+
+        stbi_image_free(pData);
+      }
+      else
+      {
+        m_loadStatus = vcSLS_Failed;
+      }
     }
 
-    vcImageRenderInfo *pImageInfo = &m_image;
-    pRenderData->images.Insert(i, &pImageInfo);
+    if (m_image.pTexture != nullptr)
+    {
+      if (m_image.type == vcIT_StandardPhoto)
+      {
+        // For now brute force sorting (n^2)
+        double distToCameraSqr = udMagSq3(m_image.position - pProgramState->pCamera->position);
+        size_t i = 0;
+        for (; i < pRenderData->images.length; ++i)
+        {
+          if (udMagSq3(pRenderData->images[i]->position - pProgramState->pCamera->position) < distToCameraSqr)
+            break;
+        }
+
+        vcImageRenderInfo *pImageInfo = &m_image;
+        pRenderData->images.Insert(i, &pImageInfo);
+      }
+      else
+      {
+        vcRenderPolyInstance *pPoly = pRenderData->polyModels.PushBack();
+        pPoly->worldMat = udDouble4x4::rotationYPR(m_image.ypr, m_image.position) * udDouble4x4::scaleUniform(m_image.scale);
+        pPoly->renderType = vcRenderPolyInstance::RenderType_Polygon;
+        pPoly->pSceneItem = this;
+        pPoly->pDiffuseOverride = m_image.pTexture;
+
+        double worldScale = vcISToWorldSize[m_image.size];
+        float aspect = 1.0f;
+        vcTexture_GetSize(m_image.pTexture, &imageSize.x, &imageSize.y);
+        aspect = float(imageSize.y) / imageSize.x;
+
+        
+        if (m_image.type == vcIT_PhotoSphere)
+        {
+          pPoly->pModel = gInternalModels[vcInternalModelType_Sphere];
+          pPoly->worldMat *= udDouble4x4::scaleUniform(worldScale);
+          pPoly->insideOut = true;
+        }
+        else if (m_image.type == vcIT_Panorama)
+        {
+          pPoly->pModel = gInternalModels[vcInternalModelType_Tube];
+          pPoly->worldMat *= udDouble4x4::scaleNonUniform(worldScale, worldScale, worldScale * aspect * UD_PI);
+          pPoly->insideOut = true;
+        }
+        else if (m_image.type == vcIT_OrientedPhoto)
+        {
+          pPoly->pModel = gInternalModels[vcInternalModelType_Quad];
+          pPoly->worldMat *= udDouble4x4::scaleNonUniform(worldScale, worldScale, worldScale * aspect) * udDouble4x4::rotationZ(UD_PI);
+          pPoly->insideOut = true;
```

Comment:
I don't think this one should be true?

Reply:
@CarlCFX I'm going to merge this- can you check if they are shown on the wrong side by default?